lib/src/main/java/com/knuddels/jtokkit/EncodingFactory.java

```diff
@@ -122,7 +122,7 @@ private static Encoding from50kParameters(
             String fileName,
             Map<String, Integer> specialTokens
     ) {
-        Pattern regex = compileRegex("'(?:[sdmt]|ll|ve|re)| ?\\p{L}+| ?\\p{N}+| ?[^\\s\\p{L}\\p{N}]+|\\s+(?!\\S)|\\s+", false);
+        Pattern regex = compileRegex("'(?:[sdmt]|ll|ve|re)| ?\\p{L}++| ?\\p{N}++| ?[^\\s\\p{L}\\p{N}]++|\\s+(?!\\S)|\\s++", false);
```

Comment:
except for `\\s+(?!\\S)` all groups should be possessive, no need to step back on match failure